src/server/ua_server_internal.h

```diff
@@ -172,6 +180,24 @@ getTypesHierarchy(UA_Nodestore *ns, const UA_NodeId *leafType, size_t leafTypeSi
                  UA_NodeId **typeHierarchy, size_t *typeHierarchySize,
                  UA_Boolean walkDownwards);
 
+
+/* Filters an event according to the filter specified by mon and then adds it to
+ * mons notification queue */
+UA_StatusCode UA_EXPORT
+UA_Event_addEventToMonitoredItem(UA_Server *server, const UA_NodeId *event, UA_MonitoredItem *mon);
+
+/* generates a unique event id */
+UA_StatusCode UA_EXPORT
+UA_Event_generateEventId(UA_Server *server, UA_ByteString *generatedId);
```

Comment:
Events automatically have a unique id when they are generated.
They are represented by a node in the information model.
So the id is automatically unique.